### PR TITLE
Extend support for Query DSL

### PR DESCRIPTION
**Pull request by bullfight**
This PR the following Query DSL functions

* `Arelastic::Queries::FunctionScore` - Function Score 
* `Arelastic::Queries::Filter` - Filter
* `Arelastic::Queries::Bool` - Adds parameter options to support `boost` and `minimum_should_match`

### CODE REVIEW

File: lib/arelastic/queries/filter.rb [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+        @query = query
+        @options = options
+      end
+
+      def as_elastic
+        params = convert_to_elastic(query)
+        { 'filter' => params }.update(options)
       end
     end
   end
```
Comment (matthuhiggins) [inline lib/arelastic/queries/filter.rb]: ✅ This one is good.

File: lib/arelastic/queries/function_score.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          searches[k] = convert_to_elastic(v) if v
+        end
+
+        opts = stringify_options(options)
+
+        { 'function_score' => searches.update(opts) }
+      end
+    end
+  end
+end
```
Comment (matthuhiggins) [inline lib/arelastic/queries/function_score.rb]: This api prevents passing strings as keys.

File: lib/arelastic/nodes/node.rb [hunk 3]
```diff
@@ -13,6 +13,10 @@ def convert_to_elastic(expr)
         end
       end
 
+      def stringify_options(options)
+        options.each_with_object({}) { |(k, v), memo| memo[k.to_s] = v }
+      end
```
Comment (matthuhiggins) [inline lib/arelastic/nodes/node.rb]: I am skeptical of this addition. Are you trying to make all calls to `as_elastic` return string keys? If so, then this must be applied to all nodes and not the ones in this PR.

File: lib/arelastic/queries/bool.rb [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
-      attr_accessor :must, :filter, :should, :must_not
-      def initialize(must: nil, filter: nil, should: nil, must_not: nil)
+      attr_accessor :must, :filter, :should, :must_not, :options
+      def initialize(must: nil, filter: nil, should: nil, must_not: nil, **options)
         @must     = must
         @filter   = filter
         @should   = should
         @must_not = must_not
+        @options  = options
       end
```
Comment (matthuhiggins) [inline lib/arelastic/queries/bool.rb]: I would rather remove the use of kwargs from this API. This addition does not support initialization with string keys.

File: lib/arelastic/nodes/node.rb (hunk 3 above)
Comment (bullfight) [reply lib/arelastic/nodes/node.rb]: This is a result of using kwags, and trying to use an `**options` which only allows for symbolized keys. If I remove the use of kwargs it shouldn't be necessary.

**Comment by bullfight** [comment]
This should be ready.

**Comment by matthuhiggins** [comment]
Thanks. Use the git ref in your branch - i'll release a gem in an hour

**Pull request merged**